### PR TITLE
Add `ALTER SOURCE xx FORMAT xx ENCODE xx (...)`

### LINKED ISSUE #1776: Document: feat: implement `ALTER SOURCE xx FORMAT xx ENCODE xx (...)`
This issue tracks the documentation update needed for the merged PR #14057.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/14057
Source PR Merged At: 2024-01-17T03:27:37Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ Add `alter source format ... encode ...` to `alter source` command.]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ https://github.com/risingwavelabs/risingwave/pull/14057/files]

- **Related doc issue**
  
  Resolves [ https://github.com/risingwavelabs/risingwave-docs/issues/1776#event-11550570033 ]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Review by Rossil2012 — COMMENTED** [review]
Please also add the statement into page `modify-schemas`. https://docs.risingwave.com/docs/current/modify-schemas/

**Review by Rossil2012 — APPROVED** [review]
The content LGTM.

File: docs/ingest/modify-schemas.md [hunk 1]
```diff
@@ -58,3 +58,48 @@ To remove a column from a table, use this command:
...
+```sql title=Example
+ALTER SOURCE src_user FORMAT PLAIN ENCODE PROTOBUF(
+    schema.registry = 'http://message_queue:8081',
+    message = 'test.UserWithMoreFields'
+);
+```
+
+
+:::note
+Altering `data_format` and `data_encode` is not supported at present. In addition, for refreshing the schema registry of a source, if the statement will drop columns or change types, it's not allowed.
```
Comment (WanYixian) [inline docs/ingest/modify-schemas.md]:
```suggestion
Currently, it is not supported to modify the `data_format` and `data_encode`. Furthermore, when refreshing the schema registry of a source, it is not allowed to drop columns or change types in the statement.
```

File: docs/sql/commands/sql-alter-source.md [hunk 2]
```diff
@@ -100,3 +100,42 @@ ALTER SOURCE current_source_name
...
+```sql title=Example
+ALTER SOURCE src_user FORMAT PLAIN ENCODE PROTOBUF(
+    schema.registry = 'http://message_queue:8081',
+    message = 'test.UserWithMoreFields'
+);
+```
+
+
+:::note
+Altering `data_format` and `data_encode` is not supported at present. In addition, for refreshing the schema registry, if the statement will drop columns or change types, it's not allowed.
```
Comment (WanYixian) [inline docs/sql/commands/sql-alter-source.md]:
```suggestion
Currently, it is not supported to modify the `data_format` and `data_encode`. Furthermore, when refreshing the schema registry of a source, it is not allowed to drop columns or change types in the statement.
```

**Review by WanYixian — APPROVED** [review]
LGTM

**Review by CharlieSYH — APPROVED** [review]

**Pull request merged**